src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -609,6 +618,9 @@ object Parameters {
     def quoteJsonFieldName: Boolean = {
       isTrue(parameters.getOrElse(PARAM_INTERNAL_QUOTE_JSON_FIELD_NAME, "true"))
     }
+    def checkTableExistenceInCurrentSchemaOnly: Boolean = {
```

Comment:
If this param is true, shouldn't we just always call ```alter session set search_path='$current'```

Reply:
It's doable. But I think it's a better choice to use a new connection.
The problem to use existing connection is that SC needs to get the current value first, and then restore it in the end. SC needs to parse the query result when retrieving the current values. If there is potential interface change, it will break SC. The criteria is lowest risk. I think we should use a new connection.